### DISCUSSION TITLE
Can't build without optional dependencies - CMake noob

### DISCUSSION THREAD

**Discussion by HenriqueCrj** [alandefreitas/matplotplusplus#196] [Q&A]
Hi, I'm using Ubuntu 20.04 LTS on WSL 2. When I try to build my project I get the following error: 

![image](https://user-images.githubusercontent.com/61526171/136325974-ee22a572-c3d6-45b0-adba-aa023270c50d.png)

TIFF is one of the optional libraries for image manipulation, but it looks like I can't build because of it.
Before this error showed up, I had the same error message with JPEG library, and after installing it, the error disapeared. Should I install all the libraries even when they are optional? What am I doing wrong?

**Comment by alandefreitas** [comment]
The configuration script looks for TIFF at this point:

https://github.com/alandefreitas/matplotplusplus/blob/c088749434154c230ee7547c6871d65e58876dc6/source/3rd_party/CMakeLists.txt#L79

and the corresponding options are only set if `TIFF_FOUND` is set.

I don't know what's happening there, but note that CMake only looks for TIFF but not `/usr/lib/x86_64-linux-gnu/libtiff.so` specifically. It seems like your configure script thinks it did find TIFF at `/usr/lib/x86_64-linux-gnu/libtiff.so` specifically for some reason.

To debug that, you could check if TIFF is being found at this point (if `TIFF_FOUND` is being defined) and at what point and how the FindTIFF.cmake script is finding it.

**Comment by HenriqueCrj** [comment]
I couldn't debug and ended up cloning the repository and adding matplotplusplus as subdirectory, it was the easiest way. I'll learn how to install a library and try the find_package later. With Windows this doesn't happen, but I have other kind of problem.
Thank you for answering my question.
